src/node/desktop/src/native/desktop/desktop.cc

```diff
@@ -197,13 +198,76 @@ Napi::Value isCtrlKeyDown(const Napi::CallbackInfo& info)
    return Napi::Boolean::From(info.Env(), value);
 }
 
+
+namespace {
+
+wchar_t* currentCSIDLPersonalHomePathImpl()
+{
+   #ifdef _WIN32
+   // query for My Documents directory
+   const DWORD SHGFP_TYPE_CURRENT = 0;
+   wchar_t homePath[MAX_PATH];
+   HRESULT hr = ::SHGetFolderPathW(nullptr,
+                                   CSIDL_PERSONAL,
+                                   nullptr,
+                                   SHGFP_TYPE_CURRENT,
+                                   homePath);
+   #else
+   wchar_t homePath[1024] = {};
+   #endif
+   return homePath;
+}
+} // end anonymous namespace
+
+Napi::Value currentCSIDLPersonalHomePath(const Napi::CallbackInfo& info)
+{
+   wchar_t* value = currentCSIDLPersonalHomePathImpl();
+   // convert wide to UTF-8
+   std::wstring_convert<std::codecvt_utf8<wchar_t>> conv1;
```

Comment:
I think it might be better to use `codecvt_utf8_utf16` although I'm not sure. `codecvt_utf8` seems to convert between UTF-8 and UTF-32. I think in testing we want to be sure that users with special characters in their username work.